File: hackerRank/algorithms/SimpleArraySum/SimpleArraySum.js
// https://www.hackerrank.com/challenges/simple-array-sum/problem
function simpleArraySum(ar) {
    // way 1
    // let total = 0;
    // ar.forEach(item => total += item);
    // return total;

    // way 2
    return ar.reduce((total, curr) => total + curr);
}

console.log(simpleArraySum([1, 2, 3]));             //6
console.log(simpleArraySum([1, 2, 3, 4, 10, 11]));  //31
console.log(simpleArraySum([5, 2, 1, 4, 73, 25]));  //110

File: hackerRank/algorithms/SimpleArraySum/SimpleArraySum.java
package algorithms.SimpleArraySum;
import java.util.Arrays;

// https://www.hackerrank.com/challenges/simple-array-sum/problem
public class SimpleArraySum  {
    static int simpleArraySum(int[] ar) {
        return Arrays.stream(ar).sum();
    }

    public static void main(String[] args) {
        System.out.println(simpleArraySum(new int[]{1, 2, 3}));             //6
        System.out.println(simpleArraySum(new int[]{1, 2, 3, 4, 10, 11}));  //31
        System.out.println(simpleArraySum(new int[]{5, 2, 1, 4, 73, 25}));  //110
    }
}
